extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php

```diff
@@ -67,16 +66,16 @@ private function getUpvoteRequestUrl() {
 		return "/$this->cityId/votes/post/";
 	}
 
-	public function getData( $showLatest, $limit, $category ) {
+	public function getData( $showLatest, $limit, $categoryId ) {
 		$sortKey = $showLatest ? self::SORT_LATEST_LINK : self::SORT_TRENDING_LINK;
-		$categoryId = false;
 		$invalidCategory = false;
 		$discussionsUrl = false;
+		$categoryName = false;
 
-		if ( !empty( $category ) ) {
-			$categoryId = $this->getCategoryId( $category );
+		if ( !empty( $categoryId ) ) {
```

Comment:
Since the discussion URL always starts with the same string, this if block could be simplified to:
```php
$discussionsUrl = "/d/f?sort=$sortKey";
if ( !empty( $categoryId ) ) {
    $categoryName = $this->getCategoryName( $categoryId );
    if ( $categoryName ) {
        $discussionsUrl .= "&catId=$categoryId";
    }
}
...
return [
    ...
    'invalidCategory' => ( $categoryId && !$categoryName ),
];
```